### ISSUE TITLE
My courses sort and size

### ISSUE THREAD

**Issue by gjb2048** [gjb2048/moodle-theme_essential#658]
From: Peter Diedrichs - https://moodle.org/user/profile.php?id=8060

My Courses menu in the header: It's good that it kan be resticted to show only open courses. 
What would make it great would be the options to: 
- Order courses by their ID (their actual obligatory ID, not the optional one in Settings) and choose "incremented" (0-9) or "decremented" (0-9) order
- Set a maximum number of courses to be shown.
Then it would become a quick menu for users with lots of courses, showing only their (for instance) last five courses.

**Comment by gjb2048** [comment]
Do you really mean "choose "incremented" (0-9) or "decremented" (9-0) order"?

**Comment by gjb2048** [comment]
Also, what is the criteria for the 'last five courses'?  Access time?

**Comment by cdipe** [comment]
Hi! (I'm not sure where my response went, I'll write a new) 
I was thinking the last courses that the user was added to. But, maybe last accessed would be better? What do you think? 
It should probably be the ten last, five might be to few. I think it should be selectable in the Essentials admin settings.

**Comment by gjb2048** [comment]
Hi @cdipe,

Pragmatically I'm not going to spend any more time / thought on this until one of two things happen:

1. Gets funded.
2. I decide in the medium to long term future that its something I want / would be good for me to do.

Kind regards,

Gareth

**Comment by cdipe** [comment]
Hi Gareth, I totally understand that. I need an approximate of the cost in order to get the funding. But I also need to know what a "cild theme" means in the long run, will it be updated along with the main theme?

**Comment by gjb2048** [comment]
With bespoke software development the approximate cost is only known at the end.  I could work out an estimated cost which would be a range.  Do you have financial authority to request and approve quotes?

A child theme is specific to you.  It would not be maintained in line with the main theme unless we had a support agreement.  However it will be small with minor additions and come with 'if I change the API within X period I'll update it'.

**Comment by cdipe** [comment]
Yes, I have the authority to request and approve a certain quote, but depending on the amount I need further authorization since this is not in my budget for 2016.


________________________________________

Peter Diedrichs

Förvaltningsledare Utbildningsmiljö

Universitetsbiblioteket

Linnéuniversitetet, Kalmar

Telefon +46(0)480-446041

Mobil -46(0)708–624567

http://lnu.se<http://lnu.se/>

Från: Gareth J Barnard <notifications@github.com<mailto:notifications@github.com>>
Svara till: gjb2048/moodle-theme_essential <reply@reply.github.com<mailto:reply@reply.github.com>>
Datum: tisdag 24 maj 2016 08:28
Till: gjb2048/moodle-theme_essential <moodle-theme_essential@noreply.github.com<mailto:moodle-theme_essential@noreply.github.com>>
Ämne: Re: [gjb2048/moodle-theme_essential] My courses sort and size (#658)

With bespoke software development the approximate cost is only known at the end. I could work out an estimated cost which would be a range. Do you have financial authority to request and approve quotes?

A child theme is specific to you. It would not be maintained in line with the main theme unless we had a support agreement. However it will be small with minor additions and come with 'if I change the API within X period I'll update it'.

**Comment by gjb2048** [comment]
Ok, Peter, do you have an eMail address I can send my terms of my terms of business to?

**Comment by cdipe** [comment]
Yes, peter.diedrichs@lnu.se



________________________________________

Peter Diedrichs

Förvaltningsledare Utbildningsmiljö

Universitetsbiblioteket

Linnéuniversitetet, Kalmar

Telefon +46(0)480-446041

Mobil -46(0)708–624567

http://lnu.se<http://lnu.se/>

Från: Gareth J Barnard <notifications@github.com<mailto:notifications@github.com>>
Svara till: gjb2048/moodle-theme_essential <reply@reply.github.com<mailto:reply@reply.github.com>>
Datum: tisdag 24 maj 2016 09:37
Till: gjb2048/moodle-theme_essential <moodle-theme_essential@noreply.github.com<mailto:moodle-theme_essential@noreply.github.com>>
Kopia: Peter Diedrichs <peter.diedrichs@lnu.se<mailto:peter.diedrichs@lnu.se>>, Comment <comment@noreply.github.com<mailto:comment@noreply.github.com>>
Ämne: Re: [gjb2048/moodle-theme_essential] My courses sort and size (#658)

Ok, Peter, do you have an eMail address I can send my terms of my terms of business to?

**Comment by gjb2048** [comment]
NTS: user_lastaccess

**Comment by gjb2048** [comment]
Ok, times (development includes testing):

Investigation: 1.5 hours.

Essential development:
Sort order setting: Course sort order or by last accessed time (Custom DB SQL on user_lastaccess) then course id.
Maximum courses setting: 0 for all or number entered.
Time: 1 - 1.5 hours.  Say 1.5 and if less then charge less.

Child theme for:
Sort order setting: Course id - ascending or descending.
Time: 45 mins.

General: Refactor 'custom_menu_courses()' to facilitate effective implementation of both - 30 mins.

Total time 4 hours 15 mins.
